### PR TITLE
Add 3d task descriptions

### PR DESCRIPTION
**Pull request by dylanebert**
Added contents of image-to-3d and text-to-3d tasks

### CODE REVIEW

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+## Use Cases
+
+Image-to-3D models can be used in a wide variety applications that require 3D, such as games, animation, design, architecture, engineering, marketing, and more.
+
+### Meshes
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:5]: we often put full names, e.g. "Generating Meshes"

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Meshes are the standard representation of 3D in industry.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:7]: Would be nice to put an image of how a mesh looks like and explain what a mesh generation model takes input and outputs output

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Meshes can be in `.obj`, `.glb`, `.stl`, or `.gltf` format. Other formats are allowed, but won't be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component.
+
+### Splats
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:11]:
```suggestion
### Generating Gaussian Splats
```

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+[Gaussian Splatting](https://huggingface.co/blog/gaussian-splatting) is a differentiable rasterization technique that represents 3D as fuzzy points.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:13]: can you explain this in simpler terms for a game dev that doesn't know much about this?

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+Splats can be in `.ply` or `.splat` format. They can be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component using the [gsplat.js](https://github.com/huggingface/gsplat.js) library.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:15]: I feel like this is a bit drowning the use case itself in general sense, you can put these details somewhere below instead

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+### Inference
+
+You can use custom diffusion pipelines to infer with `image-to-3D` models.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:19]: maybe explain that it uses diffusers library etc

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+These are unstandardized and depend on the model.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:21]: I'd add they can be seen in each model repository

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:22]: maybe go through what you're doing (creating pipeline etc)

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+input_image = np.array(input_image, dtype=np.float32) / 255.0
+result = pipeline("", input_image)
+result_path = "/tmp/output.ply"
+pipeline.save_ply(result, result_path)
+print(f"Output saved to {result_path}")
+```
+
+## Useful Resources
+
+- [ML for 3D Course](https://huggingface.co/learn/ml-for-3d-course)
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:49]: I'd add gsplat.js docs too!

File: packages/tasks/src/tasks/image-to-3d/data.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+		inputs: [
+			{
+				filename: "image-to-3d-image-input.png",
+				type: "img",
+			},
+		],
+		outputs: [
+			{
+				label: "Result",
+				content: "image-to-3d-3d-output-filename.glb",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/data.ts:24]: I don't know if we can render glb files, can you put an image of how final 3d image looks like? and maybe put in label that it's a glb file

File: packages/tasks/src/tasks/image-to-3d/data.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+				type: "text",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/data.ts:25]: this is an image I think

File: packages/tasks/src/tasks/image-to-3d/data.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+			id: "ashawkey/LGM",
+		},
+	],
+	spaces: [
+		{
+			description: "Leaderboard to evaluate image-to-3D models.",
+			id: "dylanebert/3d-arena",
+		},
+		{
+			description: "InstantMesh-based image-to-3D demo with mesh outputs.",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/data.ts:54]: what is InstantMesh?

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
+## Use Cases
+
+Text-to-3D models can be used in a wide variety applications that require 3D, such as games, animation, design, architecture, engineering, marketing, and more.
+
+This task is similar to the [image-to-3d](https://huggingface.co/tasks/image-to-3d) task, but takes text input instead of image input. In practice, this is often equivalent to a combination of [text-to-image](https://huggingface.co/tasks/text-to-image) and [image-to-3d](https://huggingface.co/tasks/image-to-3d).
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/about.md:5]: `In practice, this is often equivalent to a combination of [text-to-image](https://huggingface.co/tasks/text-to-image) and [image-to-3d](https://huggingface.co/tasks/image-to-3d).` how, as in, does it input image and text simultaneously?

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot! 🤗

File: packages/tasks/src/tasks/text-to-3d/data.ts [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+		{
+			description: "Descriptive captions for 3D objects in Objaverse.",
+			id: "tiange/Cap3D",
+		},
+	],
+	demo: {
+		inputs: [
+			{
+				label: "Prompt",
+				content: "Enter a prompt to generate a 3D model.",
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/data.ts:18]: I think you can write an example prompt here

**Comment by dylanebert** [comment]
Thanks! I have made revisions and opened a PR at https://huggingface.co/datasets/huggingfacejs/tasks/discussions/4

File: packages/tasks/src/tasks/image-to-3d/about.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    trust_remote_code=True,
+).to("cuda")
+
+input_url = "https://huggingface.co/datasets/dylanebert/iso3d/resolve/main/jpg@512/a_cat_statue.jpg"
+input_image = Image.open(BytesIO(requests.get(input_url).content))
+input_image = np.array(input_image, dtype=np.float32) / 255.0
+result = pipeline("", input_image)
+result_path = "/tmp/output.ply"
+pipeline.save_ply(result, result_path)
+print(f"Output saved to {result_path}")
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/image-to-3d/about.md:42]: no need for this line

File: packages/tasks/src/tasks/text-to-3d/about.md (hunk 13 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/text-to-3d/about.md:7]: Can you clarify?

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 16]
```diff
@@ -4,17 +4,13 @@ Text-to-3D models can be used in a wide variety applications that require 3D, su
...
 Meshes are the standard representation of 3D in industry.
 
-Meshes can be in `.obj`, `.glb`, `.stl`, or `.gltf` format. Other formats are allowed, but won't be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component.
+### Generating Gaussian Splats
 
-### Splats
-
-[Gaussian Splatting](https://huggingface.co/blog/gaussian-splatting) is a differentiable rasterization technique that represents 3D as fuzzy points.
-
-Splats can be in `.ply` or `.splat` format. They can be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component using the [gsplat.js](https://github.com/huggingface/gsplat.js) library.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/text-to-3d/about.md]: I think this can be somewhere way below but I'd still keep it really! (for both pages)

File: packages/tasks/src/tasks/text-to-3d/about.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+Meshes are the standard representation of 3D in industry.
+
+Meshes can be in `.obj`, `.glb`, `.stl`, or `.gltf` format. Other formats are allowed, but won't be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component.
+
+### Splats
+
+[Gaussian Splatting](https://huggingface.co/blog/gaussian-splatting) is a differentiable rasterization technique that represents 3D as fuzzy points.
+
+Splats can be in `.ply` or `.splat` format. They can be rendered in the gradio [Model3D](https://www.gradio.app/docs/gradio/model3d) component using the [gsplat.js](https://github.com/huggingface/gsplat.js) library.
+
```
[the remark this answers is not in the document]
Comment (merveenoyan) [reply packages/tasks/src/tasks/text-to-3d/about.md:16]: this is not addressed

File: packages/tasks/src/tasks/image-to-3d/about.md (hunk 9 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/image-to-3d/about.md:53]: this is not addressed

**Review by merveenoyan — COMMENTED** [review]
Thank you!!!

**Comment by dylanebert** [comment]
Thank you! I believe all comments are now addressed :)

**Comment by merveenoyan** [comment]
There are many comments that aren't unresolved still, can you open a follow-up PR if possible? @dylanebert
